File: The C++ Programming Language Special Edition (2000)/chapter6/exercise_01.cpp
/*
 * Exercise 1.
 *
 * Rewrite the following for statement as an equivalent while statement:
 *
 * for (i=O; i<max_length; i++) if (input_line[i] == '?') quest_count++;
 *   Rewrite it to use a pointer as the controlled variable, that is, so that 
 *   the test is of the form *p== '? ' .
 *
 */
#include <cassert>

int main()
{
	const char input[] = "test?";
	int quest_count = 0;
	const char *i = input;
	
	while(*i++) if (*i == '?') ++quest_count;
	assert(quest_count == 1);
}

File: books/Programming in Haskell/Chapter 02/exercise01.hs
{-
 - Exercise 2-1.
 -
 - 1. Parenthesise the following arithmetic expressions:
 - 2 ↑ 3 ∗ 4
 - 2 ∗ 3 + 4 ∗ 5
 - 2 + 3 ∗ 4 ↑ 5
 -}

a1 =  2 ^ 3 * 4
b1 = (2 ^ 3) * 4
r1 = a1 == b1

a2 = 2 * 3 + 4 * 5
b2 = (2 * 3) + (4 * 5)
r2 = a2 == b2

a3 = 2 + 3 * 4 ^ 5
b3 = 2 + (3 * (4 ^ 5))
r3 = a3 == b3
